pkg/ocm/versions.go

```diff
@@ -331,3 +309,53 @@ func CheckAndParseVersion(availableUpgrades []string, version string) (string, e
 	}
 	return availableUpgrades[0], nil
 }
+
+// Validate OpenShift versions
+func (c *Client) ValidateVersion(version string, versionList []string, channelGroup string, isSTS,
+	isHostedCP bool) (string, error) {
+	if version == "" {
+		return version, nil
+	}
+	// Check and set the cluster version
+	hasVersion := false
+	for _, v := range versionList {
+		if v == version {
+			hasVersion = true
+		}
+	}
+	if !hasVersion {
+		allVersions := strings.Join(versionList, " ")
+		err := fmt.Errorf("A valid version number must be specified\nValid versions: %s", allVersions)
+		return version, err
+	}
+
+	if isSTS && !HasSTSSupport(version, channelGroup) {
+		err := fmt.Errorf("Version '%s' is not supported for STS clusters", version)
+		return version, err
+	}
+	if !HasSTSSupportMinor(version) {
```

Comment:
Should we check that only for **STS** clusters?

Reply:
Though I see we didn't check if it's STS previously with that `if` statement, so I'm not sure about this one.